src/HyperfundFactory.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.28;
+
+import "./Hyperfund.sol"; // Import Hyperfund contract
+import "./Hyperstaker.sol"; // Import Hyperstaker contract
+
+contract HyperfundFactory {
+    Hyperfund[] public hyperfunds; // Array to store deployed Hyperfund contracts
+    Hyperstaker[] public hyperstakers; // Array to store deployed Hyperstaker contracts
+
+    // Mapping to associate (manager address, hypercert ID) with Hyperfund and Hyperstaker addresses
+    mapping(address => mapping(uint256 => Hyperfund)) public hyperfundsByManager;
+    mapping(address => mapping(uint256 => Hyperstaker)) public hyperstakersByManager;
```

Comment:
why are you saving these by manager?

Reply:
Coz one who is deploying is manager. I will change var to hyperfund and hyperstaker.

Reply:
also these the only reason i see to keep those is to make sure no 2 hyperfunds/hyperstakers are created for the same hypercert, so maybe add that :)

Reply:
and you can change this mapping to only keep booleans and not addresses

Reply:
and update the doc string :)

Reply:
We will need to keep address in order to fetch the address of pool by id within contract

Reply:
why do you need to fetch it onchain?

Reply:
We might need in the future if we want to create something like router to move funds/hypercerts between pools. Hence better to have this mapping

Reply:
you mean move hypercerts between hyperfunds? that doesn't make sense.. and we can have this off chain, no need for a mapping onchain to use that information